src/main/scala/br/unb/cic/oberon/cfg/ControlFlowGraphBuilder.scala

```diff
@@ -11,7 +11,16 @@ import scala.collection.mutable.Stack
 trait GraphNode
 
 case class StartNode() extends GraphNode
-case class SimpleNode(stmt: Statement) extends GraphNode
+case class SimpleNode(stmt: Statement) extends GraphNode {
+
+  override def equals(arg: Any) : Boolean = {
+    arg match {
+      case null => false
+      case SimpleNode(argStmt) => stmt.eq(argStmt)
+      case _ => false
+    }
+  }
+}
```

Comment:
I am not completely whether this is really necessary. Case classes automatically derive an implementation for the equals method.